bolt-taproot-gossip.md

```diff
@@ -0,0 +1,1083 @@
+protocol, the following scenarios may exist for any node on the network:
+
+| scenario | has legacy channels | has taproot channels  | should send `node_announcement` | should send `node_announcement_2` |
+|----------|---------------------|-----------------------|---------------------------------|-----------------------------------|
+| 1        | no                  | no                    | no                              | no                                |
+| 2        | yes                 | no                    | yes                             | no                                |
+| 3        | yes                 | yes                   | yes                             | no                                |
+| 4        | no                  | yes                   | no                              | yes                               |
+
+### Scenario 1
+
+These nodes have no announced channels and so should not be broadcasting legacy
+or new node announcement messages.
+
+### Scenario 2
+
+If a node has legacy channels but no taproot channels, they should broadcast
+only a legacy `node_announcement` message. Both taproot-gossip aware and unaware
+nodes will be able to verify the node announcement since both groups are able to
+process the `channel_announcement` for the legacy channel(s). The reason why 
+upgraded nodes should continue to broadcast the legacy announcement is because 
+this is the most effective way of spreading the `option_taproot_gossip` feature 
+bit since un-upgraded nodes can assist in spreading the message. 
+
+### Scenario 3
+
+If a node has both legacy channels and taproot channels, then like Scenario 2 
+nodes, they should continue broadcasting the legacy `node_announcement` 
+message.
+
+### Scenario 4
+
+In this scenario, a node has only taproot channels and no legacy channels. This
+means that they cannot broadcast a legacy `node_announcement` message since 
+un-upgraded nodes will drop this message due to the fact that no legacy 
+`channel_announcement` message would have been received for that node. These 
+nodes will also not be able to use a legacy `node_announcement` to propagate 
+their new `node_announcement_2` message.
+
+### Consideration & Suggestions
+
+While the network is in the upgrade phase, the following suggestions apply:
+
+- Keeping at least a single, announced, legacy channel open during the initial 
+  phase of the transition to the new gossip protocol could be very beneficial 
+  since nodes can continue to broadcast their legacy `node_announcement` and 
+  thus more effectively advertise their support for `option_taproot_gossip`. 
+- Nodes are encouraged to actively connect to other nodes that advertise the 
+  `option_taproot_gossip` feature bit as this is the only way in which they 
+  will learn about taproot channel announcements and updates.
+- Nodes should not use the new `node_announcement_2` message until they have
```

Comment:
What's the rationale here? If they don't use `node_announcement_2` at all, then none of their taproot channels will propagate.